data-model/src/main/java/org/hypertrace/core/datamodel/shared/DataFlowMetrics.java

```diff
@@ -0,0 +1,23 @@
+package org.hypertrace.core.datamodel.shared;
+
+/**
+ * Set of metrics that can be used to track the flow of data from and its processing times as it passes through various
+ * components.
+ */
+public enum DataFlowMetrics {
+  CREATION_TIME("creation.time"),
+  ENRICHMENT_ARRIVAL_TIME("enrichment.arrival.time"),
+  ENRICHMENT_ARRIVAL_LAG("enrichment.arrival.lag"),
+  VIEW_GENERATION_ARRIVAL_TIME("view.generation.arrival.time"),
+  VIEW_GENERATION_ARRIVAL_LAG("view.generation.arrival.lag");
```

Comment:
@samarth-gupta-traceable I'm not sure if we want to keep the lag metrics in here because they're specific to each job and they should be able to report as whatever metric each job wants as.

Reply:
done